packages/react-native/React/Views/RCTViewManager.m

```diff
@@ -218,12 +218,18 @@ - (RCTShadowView *)shadowView
 
 RCT_CUSTOM_VIEW_PROPERTY(transform, CATransform3D, RCTView)
 {
-  view.layer.transform = json ? [RCTConvert CATransform3D:json] : defaultView.layer.transform;
+  view.rawTransform = json;
+  view.layer.transform = json ? [RCTConvert CATransform3D:view.rawTransform viewWidth:view.bounds.size.width viewHeight:view.bounds.size.height transformOrigin: view.transformOrigin] : defaultView.layer.transform;
```

Comment:
Is this so you can use `%`s in the transform property?

Reply:
yeah exactly, also `top`, `left` etc enums require height and width of the view.